src/state/opamPackageVar.ml

```diff
@@ -305,6 +306,19 @@ let resolve st ?opam:opam_arg ?(local=OpamVariable.Map.empty) v =
        with Not_found -> Some (string ""))
     | "dev", Some opam -> Some (bool (is_dev_package st opam))
     | "build-id", Some opam -> OpamStd.Option.map string (build_id st opam)
+    | "opamfile-loc", Some opam ->
+      let repos_roots reponame =
+        match Hashtbl.find st.switch_repos.repos_tmp reponame with
+        | lazy repo_root -> repo_root
+        | exception Not_found ->
+          OpamRepositoryPath.root st.switch_global.root reponame
+      in
+      OpamFile.OPAM.get_metadata_dir ~repos_roots opam
+      |> OpamStd.Option.map (fun d ->
+          OpamFilename.Op.(d//"opam")
```

Comment:
Hm, does this work for pinned packages ? Can't the file be named differently?

Reply:
for pinned packages, it's stored in the `overlay` dir, in the [file `opam`](https://github.com/ocaml/opam/blob/master/src/format/opamPath.ml#L254), and from repo, its the package dir with name `opam`.

Reply:
aah right, it'll never get back to the original srcdir, we always have a mirror, got it! :+1:
maybe worth adding a comment ?

Reply:
It's always there, in the current run of opam (repo dir are tmp ones), yes. In fact, it's the code of `OpamRepositoryState.get_root`, that is not reachable from `OpamPackageVar`.
+1 for the comment